rasa/core/policies/policy.py

```diff
@@ -272,40 +309,30 @@ def _featurize_for_prediction(
             == SupportedData.ML_DATA,
         )
 
-    def train(
-        self,
-        training_trackers: List[TrackerWithCachedStates],
-        domain: Domain,
-        interpreter: NaturalLanguageInterpreter,
-        **kwargs: Any,
-    ) -> None:
-        """Trains the policy on given training trackers.
+    @abc.abstractmethod
+    def train(self, **kwargs: Any,) -> Resource:
```

Comment:
I've just thought don't all policies take `domain` and `training_trackers`?
If so can this be:

```
def train(
     self,
     training_trackers: List[TrackerWithCachedStates],
     domain: Domain,
     **kwargs: Any,
) -> Resource
```

Reply:
they do now, but maybe one in the future doesn't. I'd keep it fully flexible for now.

Reply:
Yeah, it gives us the flexibility to add extra data but also removes the flexibility to remove parameters.
E.g. we once had a `MappingPolicy` which only required the domain https://github.com/RasaHQ/rasa/blob/2.8.x/rasa/core/policies/mapping_policy.py

Reply:
I think if we ever change the parameters drastically (or re-introduce policies that really need no training), then we should to think about new interfaces anyway, don't we? (because e.g. `MappingPolicy` effectively didn't need any training at all and would never appear in our graphs with a train step and would never require training data - it does still require the whole conversation / trackers for inference though)

Reply:
@aeshky , what do you think?

Reply:
Added them to `train` and `predict_action_probabilities` 👍🏻